extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -539,11 +539,13 @@ export class AmpSelector extends AMP.BaseElement {
    * @private
    */
   clearSelection_(element) {
-    element.removeAttribute('selected');
-    element.setAttribute('aria-selected', 'false');
-    const selIndex = this.selectedElements_.indexOf(element);
-    if (selIndex !== -1) {
-      this.selectedElements_.splice(selIndex, 1);
+    if (element !== undefined) {
```

Comment:
We typically use guard clauses instead of branching. e.g.

```js
if (!element) {
  return;
}

/* do things now that we know element is defined */
```

This reduces indentation, and makes the code easier to read.

Reply:
Also, the `element` parameter is annotated above to always be a defined, non-nullable `Element` so somewhere the value passed in is incorrect. Instead of adding a guard to this method, we should fix the callsite that passes in the incorrect value.

Reply:
Ah, ok. Made the modifications you recommended. For future reference, if the element parameter is annotated to always be defined, why doesn't clearSelection error when called with undefined? I'd put the guard in the function since it might be called improperly even though it is specified not to.

Reply:
I think Closure Compiler makes the assumption that all values coming from an `Array<!Element>` are of type `!Element` and it also assumes that the array will be accessed from within its bounds. So because Closure doesn't guarantee that Arrays will always be accessed within their bounds, it can't warn us about this mistake.